server/bin/pbench-dispatch.py

```diff
@@ -0,0 +1,405 @@
+#!/usr/bin/env python3
+# -*- mode: python -*-
+
+
+""" 
+# check directory where tarballs are there
+
+# copy tarballs to archive area
+
+# set state to know what processing needs to be done to each tarball. it is done by 
+# setting symlinks in designated directories
+
+# those directories scanned by corresponding scripts
+
+# the work that pbench-dispatch is left for is only to move symlinks from TODO to 
+# TO-UNPACK and TO-BACKUP
+
+# then pbench-unpack-tarball will shift to UNPACKED, TO_INDEX, and any other
+# pbench-unpack script can set the state for first the state which need tarballs itself
+# and then set the state for the one which need unpacked tarballs 
+"""
```

Comment:
I'd rename the script to `pbench-dispatch-unpack.py` and not refer to the old names any more. Something like this perhaps:

> This script scans the archive for tar balls in TODO state. For each one, it adds it to the state directories specified in the config option "dispatch-states", unpacks the tarball and after unpacking it also adds it to the state directories specified in the config option "unpack-states". That way, other scripts that operate on the tarball itself can start work immediately: those scripts are the ones that check the state directories specified in "dispatch-states" (currently, the only such state is TO-BACKUP); scripts that depend on the tarball being unpacked, will have to wait until this script unpacks the tar ball and places it in the proper states before they can process it (currently, these are the TO-INDEX and TO-COPY-SOSREPORTS states). For accounting purposes, we also set the state to UNPACKED: a final state which does not trigger any further processing.
```